# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.cpp
# Commit: 6c61c412fe0b
# Full Hash: 6c61c412fe0b1afbec5d58dd6fea7c76d0531ff6
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Don't use DOM objects internally in Web Codecs implementation. r=chunmin
#   
#   Depends on D181328
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D182476
# ==============================================================================

diff -r 7dd351fce5ad -r 6c61c412fe0b dom/media/webcodecs/WebCodecsUtils.cpp
--- a/dom/media/webcodecs/WebCodecsUtils.cpp	Thu Jun 29 22:33:14 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.cpp	Thu Jun 29 22:33:15 2023 +0000
@@ -105,20 +105,21 @@
       JS_GetTypedArrayByteOffset(obj), JS_GetTypedArrayByteLength(obj));
 }
 
-Result<OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer, nsresult>
-CloneBuffer(
+Result<Ok, nsresult> CloneBuffer(
     JSContext* aCx,
-    const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aBuffer) {
+    OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aDest,
+    const OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer& aSrc) {
   std::tuple<JS::ArrayBufferOrView, size_t, size_t> info =
-      GetArrayBufferInfo(aCx, aBuffer);
-  const JS::ArrayBufferOrView& bufOrView = std::get<0>(info);
+      GetArrayBufferInfo(aCx, aSrc);
+  JS::Rooted<JS::ArrayBufferOrView> abov(aCx);
+  abov.set(std::get<0>(info));
   size_t offset = std::get<1>(info);
   size_t len = std::get<2>(info);
-  if (NS_WARN_IF(!bufOrView)) {
+  if (NS_WARN_IF(!abov)) {
     return Err(NS_ERROR_UNEXPECTED);
   }
 
-  JS::Rooted<JSObject*> obj(aCx, bufOrView.asObject());
+  JS::Rooted<JSObject*> obj(aCx, abov.asObject());
   JS::Rooted<JSObject*> cloned(aCx,
                                JS::ArrayBufferClone(aCx, obj, offset, len));
   if (NS_WARN_IF(!cloned)) {
@@ -126,11 +127,10 @@
   }
 
   JS::Rooted<JS::Value> value(aCx, JS::ObjectValue(*cloned));
-  OwningMaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer clonedBufOrView;
-  if (NS_WARN_IF(!clonedBufOrView.Init(aCx, value))) {
+  if (NS_WARN_IF(!aDest.Init(aCx, value))) {
     return Err(NS_ERROR_UNEXPECTED);
   }
-  return clonedBufOrView;
+  return Ok();
 }
 
 /*